src/core/thread/mle_router.cpp

```diff
@@ -2724,7 +2724,8 @@ void MleRouter::HandleChildUpdateResponse(RxInfo &aRxInfo)
     Child *    child;
     uint16_t   addressRegistrationOffset = 0;
 
-    if ((aRxInfo.mNeighbor == nullptr) || IsActiveRouter(aRxInfo.mNeighbor->GetRloc16()))
+    if ((aRxInfo.mNeighbor == nullptr) || IsActiveRouter(aRxInfo.mNeighbor->GetRloc16()) ||
+        !Get<ChildTable>().Contains(*aRxInfo.mNeighbor))
```

Comment:
Is it possible for a neighbor to be both not in child table and not an active router?

Reply:
It is very unlikely, but I think it may be technically possible for nighbor to be `Parent` or `ParentCandidate` which were REED.

Adding the `Get<ChildTable>().Contains()` is just adding safety check to ensure that we never cast a non `Child`  object to `Child`.